artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java

```diff
@@ -158,7 +160,12 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                }
                stompMaxFramePayloadLength = stompMaxFramePayloadLength != -1 ? stompMaxFramePayloadLength : TransportConstants.DEFAULT_WEB_SOCKET_MAX_FRAME_PAYLOAD_LENGTH;
                int webSocketMaxFramePayloadLength = ConfigurationHelper.getIntProperty(TransportConstants.WEB_SOCKET_MAX_FRAME_PAYLOAD_LENGTH, -1, nettyAcceptor.getConfiguration());
-               ctx.pipeline().addLast("websocket-handler", new WebSocketServerHandler(websocketSubprotocolIds, webSocketMaxFramePayloadLength != -1 ? webSocketMaxFramePayloadLength : stompMaxFramePayloadLength));
+               String encoderConfigType = ConfigurationHelper.getStringProperty(TransportConstants.WEB_SOCKET_ENCODER_TYPE, TransportConstants.DEFAULT_WEB_SOCKET_ENCODER_TYPE, nettyAcceptor.getConfiguration());
+               WebSocketFrameEncoderType encoderType = WebSocketFrameEncoderType.valueOfType(encoderConfigType);
+               if (encoderType == null) {
```

Comment:
As noted below I'd just have the 'valuefType' method validate and throw instead of needing to check here